_inc/lib/class.jetpack-photon-image-sizes.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * The Image Sizes library.
+ *
+ * @package jetpack
+ */
+
+jetpack_require_lib( 'class.jetpack-photon-image' );
+
+/**
+ * Class Jetpack_Photon_ImageSizes
+ *
+ * Manages image resizing via Jetpack CDN Service.
+ */
+class Jetpack_Photon_ImageSizes {
+
+	/**
+	 * @var array $data Attachment metadata.
+	 */
+	public $data;
+
+	/**
+	 * @var Image Image to be resized.
+	 */
+	public $image;
+
+	/**
+	 * @var null|array $sizes Intermediate sizes.
+	 */
+	public static $sizes = null;
+
+	/**
+	 * Construct new sizes meta
+	 *
+	 * @param int   $attachment_id Attachment ID.
+	 * @param array $data          Attachment metadata.
+	 */
+	public function __construct( $attachment_id, $data ) {
+		$this->data  = $data;
+		$this->image = new Jetpack_Photon_Image( $data, get_post_mime_type( $attachment_id ) );
+		$this->generate_sizes();
+	}
+
+	/**
+	 * Generate sizes for attachment.
+	 *
+	 * @return array Array of sizes; empty array as failure fallback.
+	 */
+	protected function generate_sizes() {
+
+		// There is no need to generate the sizes a new for every single image.
+		if ( null !== self::$sizes ) {
+			return self::$sizes;
+		}
+
+		/*
+		 * The following logic is copied over from wp_generate_attachment_metadata
+		 */
+		$_wp_additional_image_sizes = wp_get_additional_image_sizes();
+
+		$sizes = array();
+
+		/*
+		 * Remove filter preventing WordPress from reading the sizes, it's meant
+		 * to prevent creation of intermediate files, which are not really being used.
+		 */
+		remove_filter( 'intermediate_image_sizes', 'wpcom_intermediate_sizes' );
```

Comment:
Should we add a check here, since `wpcom_intermediate_sizes` is wpcom-specific? This would avoid warnings like this one:
`PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, function 'wpcom_intermediate_sizes' not found or invalid function name in /wp-includes/class-wp-hook.php on line 286`

Reply:
Thanks, good catch! I have removed the addition and removal of the filter in one of the latest commits because they don't do anything anyway.